Automattic/wp-calypso
# Newsletters: Add goals screen to onboarding

## Proposed Changes

- Adds a newsletter goals screen to the newsletter onboarding flow. See mockup below from Figma.
- Choosing the 'paid' or 'import' options will set a site goal of 'paid-subscribers' or 'import-subscribers' respectively.
- Choosing paid will show Launchpad tasks for Setup a payment method and Add a paid newsletter.
- Choosing import will show a Launchpad for Add subscribers.

**Note on Add Subscribers Screen.** For now, we will still be showing the Add Subscribers screen before Launchpad. In a separate PR, we will remove that screen for all flows. Users who select the Import option will see the option to Add Subscribers on Launchpad. As part of that PR, we also need to ensure the Add Subscribers task is shown as incomplete when it first loads (it's initial state is currently complete.

**Note on Paid checkbox on Setup screen.** For now, we will still show the option for paid newsletter on the Setup screen. That will also be removed in a separate PR, which is ready [here](https://github.com/Automattic/wp-calypso/pull/79265). 

<img width="1180" alt="newsletter-goals" src="https://github.com/Automattic/wp-calypso/assets/21228350/15898d3c-420b-456e-901b-918f10921725">

## Testing Instructions

We are testing two things: first that the new screen loads and works as expects and, second, that the task list on Launchpad adjusts as expected for each goal.

1) Check out this branch and start a newsletter site at http://calypso.localhost:3000/setup/newsletter/intro
2) Confirm that you see the new Newsletter Goals screen after setup, and that the look/style matches the screenshot above. 
3) Test each goal option.
   - Click the free option, go to Launchpad, and confirm there are just two tasks (Personalize, Choose Plan) plus the Write Post button, and confirm the Add Subscribers step does not show. 
   - Click the import option, go to Launchpad and confirm there are three tasks (Personalize, Choose Plan, Add Subscribers) plus the Write button. 
   - Click the paid option, go to Launchpad, and confirm there are four tasks (Personalize, Choose Plan, Payment Method, Create Paid Newsletter), plus the write button. 



## Pre-merge Checklist
